### PR TITLE
[14.0][ADD] hr_expense_payment_widget_amount

### PR DESCRIPTION
**Pull request by Saran440**
This module adds a popover widget in the expense report view
when there are outstanding payments that allows the user.

you can link expense report > payment
Selection_001

Note: if you paid by company, you can't unreconcile payment from view expense report.

TODO:
- [x] depends on hr_expense_payment https://github.com/OCA/hr-expense/pull/55

### CODE REVIEW

File: hr_expense_payment_widget_amount/models/hr_expense_sheet.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        compute="_compute_payments_widget_reconciled_info",
+        groups="account.group_account_invoice,account.group_account_readonly",
+    )
+
+    def _get_expense_reconciled_info_JSON_values(self):
+        self.ensure_one()
+        company_currency = self.env.company.currency_id
+        content = self.account_move_id._get_reconciled_info_JSON_values()
+        # Paid by company can't reconciled because there is not account move.
+        if self.payment_mode == "company_account":
```
Comment (kittiu) [inline hr_expense_payment_widget_amount/models/hr_expense_sheet.py]:
```suggestion
        if self.account_move_id.journal_id == self.bank_journal_id:
```

File: hr_expense_payment_widget_amount/models/hr_expense_sheet.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        "payment_id": payment.id,
+                        "account_payment_id": payment.id,
+                        "payment_method_name": payment.payment_method_id.name
+                        if payment.journal_id.type == "bank"
+                        else None,
+                        "move_id": payment.move_id.id,
+                        "ref": payment.ref,
+                    }
+                )
+        elif self.payment_mode == "own_account":
```
Comment (kittiu) [inline hr_expense_payment_widget_amount/models/hr_expense_sheet.py]:
```suggestion
        elif self.account_move_id.journal_id == self.journal_id:
```

**Comment by kittiu** [comment]
@Saran440 I think this module name can be shorten a bit, `hr_expense_payment_widget`

WDYT?

**Comment by Saran440** [comment]
I have idea from https://github.com/OCA/account-payment/tree/12.0/account_payment_widget_amount

**Comment by kittiu** [comment]
Ok

**Comment by kittiu** [comment]
Please check travis

**Comment by Saran440** [comment]
@kittiu This PR travis is red because it's depend on #55

Should we combine `hr_expense_payment_widget_amount` and `hr_expense_payment` into one module?

**Comment by hitrosol** [comment]
I tested in runboat. I found that if I do the register payment in multi lines expense report, then the payment widget amount will show multiple time as per line expenses, even thought I only do one payment.
I'm afraid if we have 15 lines of expense item, then the payment widget amount will be 15 lines as well.

**Comment by Saran440** [comment]
@hitrosol
Yes, I agree it's a good point,
but in core Odoo, the expense will be paid one line at a time till it is gone.

I'm not sure we should do following standard or fix it.

have you any idea? @kittiu

**Comment by kittiu** [comment]
For usability, we should group it per say. Is it possible to fix?

**Comment by Saran440** [comment]
@hitrosol @kittiu Fixed widget show following payments. Could you review please.

**Comment by hitrosol** [comment]
I see one line payment whenever I checked the Group Payment eventhough i have multi line expense --> OK.
I try to click the view button, and it brings me to the Account Payment form --> OK.
But, whenever I click the Unreconcile, it doesn't unreconcile the payment of the expense sheet.
gambar

Maybe this can be merged at the moment and fix that thing on the next PR ?

**Comment by Saran440** [comment]
Thank you for your review.
Fixed code with popover widget document paid by employee only because paid by company will not create payment from https://github.com/OCA/hr-expense/pull/113

This PR you can unreconciled with multi lines expense.

**Comment by hitrosol** [comment]
Nice work, functionally okay from my side. No Code Review.
I hope Odoo SA can put this fuctionality in the core in future, because it was not easy to find the relation between the Expense Report to the Payment.

**Review by hitrosol — APPROVED** [review]

**Review by ps-tubtim — APPROVED** [review]
LGTM

**Pull request merged**